code/weapon/beam.cpp

```diff
@@ -2201,8 +2201,12 @@ void beam_aim(beam *b)
 		if (!(b->flags & BF_IS_FIGHTER_BEAM))
 			b->subsys->turret_next_fire_pos = b->firingpoint;
 
-		// where the shot is originating from (b->last_start gets filled in)
-		beam_get_global_turret_gun_info(b->objp, b->subsys, &b->last_start, &temp, 1, &p2, (b->flags & BF_IS_FIGHTER_BEAM) > 0);
+		if (b->subsys->system_info->flags2 & MSS_FLAG2_SHARE_FIRE_DIRECTION) {
+			beam_get_global_turret_gun_info(b->objp, b->subsys, &b->last_start, &temp, 0, NULL, (b->flags & BF_IS_FIGHTER_BEAM) > 0);
```

Comment:
I know that this is just copied from existing code, but the b->flags & SOMETHING > 0 comparison is dangerous.  b->flags is an int, and when the 32nd bit flag is implemented it'll return a negative number. Sure, that won't affect this comparison since it's only comparing the 8th bit, but I'd feel more comfortable not propagating this around the code :)

probably should be ``` != 0``` instead (assuming we're moving away from relying on implicit conversions to boolean in comparisons)